libs/stats/odc/stats/_gm_ml_pred.py

```diff
@@ -0,0 +1,545 @@
+        f'{protocol}{s3bucket}/{ref_folder}/rainfall/CHPclim_jul_dec_cumulative_rainfall.tif'
+    )
+    # chirps_paths = (
+    #     "s3://deafrica-input-datasets/rainfall/CHPclim_jan_jun_cumulative_rainfall.tif",
+    #     "s3://deafrica-input-datasets/rainfall/CHPclim_jul_dec_cumulative_rainfall.tif"
+    # )
+    rename_dict = {  # "nir_1": "nir",
+        "B02": "blue",
+        "B03": "green",
+        "B04": "red",
+        "B05": "red_edge_1",
+        "B06": "red_edge_2",
+        "B07": "red_edge_3",
+        "B08": "nir",
+        "B8A": "nir_narrow",
+        "B11": "swir_1",
+        "B12": "swir_2",
+        "BCMAD": "bcdev",
+        "EMAD": "edev",
+        "SMAD": "sdev",
+    }
+    training_features = [
+        "red_S1",
+        "blue_S1",
+        "green_S1",
+        "nir_S1",
+        "swir_1_S1",
+        "swir_2_S1",
+        "red_edge_1_S1",
+        "red_edge_2_S1",
+        "red_edge_3_S1",
+        "edev_S1",
+        "sdev_S1",
+        "bcdev_S1",
+        "NDVI_S1",
+        "LAI_S1",
+        "MNDWI_S1",
+        "rain_S1",
+        "red_S2",
+        "blue_S2",
+        "green_S2",
+        "nir_S2",
+        "swir_1_S2",
+        "swir_2_S2",
+        "red_edge_1_S2",
+        "red_edge_2_S2",
+        "red_edge_3_S2",
+        "edev_S2",
+        "sdev_S2",
+        "bcdev_S2",
+        "NDVI_S2",
+        "LAI_S2",
+        "MNDWI_S2",
+        "rain_S2",
+        "slope",
+    ]
+    resolution = (-10, 10)
+    # the time actually is the time range, required by datacube query
+    # the datetime_range is required by OutputProduct of odc-stats model
+    time = ("2019-01", "2019-12")
+    datetime_range = DateTimeRange(time[0], "P12M")
+    output_crs = "epsg:6933"
+    # query is required by open datacube
+    query = {
+        "time": time,
+        "resolution": resolution,
+        "output_crs": output_crs,
+        "group_by": "solar_day",
+    }
+    # the prd_properties is required by the stac json
+    prd_properties = {
+        "odc:file_format": "GeoTIFF",
+        "odc:producer": "digitalearthafrica.org",
+        "odc:product": f"{PRODUCT_NAME}",
+        "proj:epsg": 6933,
+        "crop-mask-model": model_path,
+    }
+    # the OutputProduct is required by stac json
+    product = OutputProduct(
+        name=PRODUCT_NAME,
+        version=PRODUCT_VERSION,
+        short_name=PRODUCT_NAME,
+        location=f'{protocol}{s3bucket}/',
+        properties=prd_properties,
+        measurements=("mask", "prob", "filtered"),
+        href=f"https://explorer.digitalearth.africa/products/{PRODUCT_NAME}",
+    )
+
+
+def down_scale_gm_band(
+        ds: xr.Dataset, exclude: Tuple[str, str] = ("sdev", "bcdev"), scale=10_000
+) -> xr.Dataset:
+    """
+    down scale band not in exclude list.
+    :param ds:
+    :param exclude:
+    :param scale:
+    :return:
+    """
+    for band in ds.data_vars:
+        if band not in exclude:
+            ds[band] = ds[band] / scale
```

Comment:
not sure it's a good idea to modify `ds` object in place.

Reply:
I will discuss it with Chad